controllers/healthcheck_controller.go

```diff
@@ -321,9 +459,25 @@ func (r *HealthCheckReconciler) watchWorkflowReschedule(ctx context.Context, log
 				hc.Status.FailedCount++
 				hc.Status.Total = hc.Status.SuccessCount + hc.Status.FailedCount
 				hc.Status.LastFailedWorkflow = wfName
-				metrics.MonitorError.With(prometheus.Labels{"healthcheck_name": hc.GetName()}).Inc()
-				metrics.MonitorStartedTime.With(prometheus.Labels{"healthcheck_name": hc.GetName()}).Set(float64(then.Unix()))
-				metrics.MonitorFinishedTime.With(prometheus.Labels{"healthcheck_name": hc.GetName()}).Set(float64(now.Time.Unix()))
+				metrics.MonitorError.With(prometheus.Labels{"healthcheck_name": hc.GetName(), "workflow": "healthcheck"}).Inc()
+				metrics.MonitorStartedTime.With(prometheus.Labels{"healthcheck_name": hc.GetName(), "workflow": "healthcheck"}).Set(float64(then.Unix()))
+				metrics.MonitorFinishedTime.With(prometheus.Labels{"healthcheck_name": hc.GetName(), "workflow": "healthcheck"}).Set(float64(now.Time.Unix()))
+				log.Info("Creating Workflow", "namespace", wfNamespace, "generateNamePrefix", hc.Spec.RemedyWorkflow.GenerateName)
+				err := r.createRBACForWorkflow(log, hc, "remedy")
+				if err != nil {
+					log.Error(err, "Error creating RBAC Permissions for Remedy Workflow")
+					break
+				}
+				generatedWfName, err := r.createSubmitRemedyWorkflow(ctx, log, hc)
+				if err != nil {
+					log.Error(err, "Error creating or submitting remedyworkflow")
+				}
+				r.watchRemedyWorkflow(log, wfNamespace, generatedWfName, hc)
```

Comment:
Is this a synchronous watch? So, when the control comes out of `r.watchRemedyWorkflow`, is it guaranteed that the workflow has completed it's execution?

Reply:
Yes.. it will return only if it is either completed successfully/ Failed / Job deleted with activedeadlineseconds which is equal to repeatAftersec.

Reply:
I fixed this one.. now the function returns and error if there is an error.